### PR TITLE
fix: add driver's preferred contact email in delivery trip

### PR DESCRIPTION
**Pull request by ParthKharwar**
[TASK-2019-00823](https://digithinkit.global/desk#Form/Task/TASK-2019-00823)

- Driver's email added to Delivery Trip DocType as it did not exist.
- Additional functions added to check and set the preferred email.
- Patch added to check and set preferred email for all existing drivers.

### CODE REVIEW

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- erpnext/patches/v12_0/set_employee_preferred_emails.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/erpnext/erpnext/pullRequest?prid=4612158&bid=15471832)

**Comment by Alchez** [comment]
@ParthKharwar, you'll have to make these changes to upstream as well. Since this is an enhancement, Frappe will only accept it in develop.

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- erpnext/patches/v12_0/set_employee_preferred_emails.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/erpnext/erpnext/pullRequest?prid=4612158&bid=15471832)

File: erpnext/patches.txt [hunk 1]
```diff
@@ -614,3 +614,4 @@ erpnext.patches.v11_1.set_status_for_material_request_type_manufacture
 erpnext.patches.v12_0.set_lead_title_field
 erpnext.patches.v11_1.set_produced_qty_field_in_sales_order_for_work_order
 erpnext.patches.v12_0.add_default_birthday_email_template
+erpnext.patches.v12_0.set_employee_preferred_emails #
```
Comment (Alchez) [inline erpnext/patches.txt]: Remove the #

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- erpnext/patches/v12_0/set_employee_preferred_emails.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/erpnext/erpnext/pullRequest?prid=4612158&bid=15471832)

**Review by Alchez — APPROVED** [review]

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- erpnext/patches/v12_0/set_employee_preferred_emails.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/erpnext/erpnext/pullRequest?prid=4612158&bid=15471832)

**Pull request merged**